mrc-ide/mint
# mrc-2492 Implement summary table for strategising across regions

Changes:
- Add strategise page: budget input form and summary table ([mockups](https://imperiallondon.sharepoint.com/:w:/r/sites/MRCGIDA-mint-dev/Shared%20Documents/mint-dev/Optimisation/Stategize%20across%20regions%20-%20mockups.docx?d=w989d5cc08537432287ec7ebe14a912b2&csf=1&web=1)). Details charts/table to follow in mrc-2493 and mrc-2494.
- Apply limit of 15 regions to place an upper bound on time taken in optimiser
- Add mutation/action to clear API errors - previously these just accumulated
- Remove vestigial(?) `package-lock.json`

To test:
- Visit localhost:8080/?stratAcrossRegions=true
- Add at least two regions, each of which has at least one intervention configured i.e. >0% ITN and/or IRS (the optimiser will work with no interventions but will return very uninteresting results)
- Click on "Strategize across regions" in the toolbar

Notes:
- There is definitely scope to turn the strategies table into a more metadata-driven component. In particular the display options  parallel some of the dynamic chart configuration we use in the rest of the app. This wouldn't be trivial though, especially dealing with a dynamic number of columns. At the same time it might make sense to also turn use vue-dynamic-form - especially if we were to add an inline (i.e. horizontal layout). Keen to hear thoughts on this - would have been hard to do speculatively but may merit a follow-up ticket.
- There is now something of a disconnect between the per-region part of the app and the new page which operates across regions. In particular the global budget needs to be accessible to each region because it is used in a formula in the graph metadata. This necessitates copying the budget into each region's settings when it changes. This also means that there is part of the settings that isn't derived directly from the dynamic forms used to otherwise configure each region. I can't see a way to resolve this without a more significant restructure of the app. If the dynamic form allowed hidden fields then we could somehow link one with a reference to a setting elsewhere in the store/state, but that's not ideal.
- There are a couple of usability issues remaining (now noted on top-level ticket mrc-2434):
  - There has never been a project page, only regions. So to strategise for a project you first have to select a region, which enables the strategise tab if the implicitly-selected project has more than one region
  - If you do create many regions (e.g. >5) then the strategies table can get very crowded
- Feature flag will be removed in mrc-2505
- This branch depends on mrc-ide/mintr#59

To-do:
- [x] Revert to mintr master
